source/FreeRTOS_IP.c

```diff
@@ -839,10 +904,7 @@ void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
         }
         else
         {
-            /* Set the actual packet size in case a bigger buffer was returned. */
-            pxNetworkBuffer->xDataLength = sizeof( UDPPacket_t ) + uxRequestedSizeBytes;
-            /* Skip 3 headers. */
-            pvReturn = &( pxNetworkBuffer->pucEthernetBuffer[ sizeof( UDPPacket_t ) ] );
+            pvReturn = NULL;
```

Comment:
We tend to initialise the return variable to the most likely outcome:
~~~
    void * pvReturn = NULL;
~~~

This avoids the need for llines like this:
~~~
            *pucIPType = ucIPType;
        }
        else
        {
            pvReturn = NULL;
        }
    }
    else
    {
        pvReturn = NULL;
    }
~~~
and also it avoids that `pvReturn` will be used without initialising it.

Reply:
Thanks, will update that.